vcd/structure.go

```diff
@@ -200,6 +203,42 @@ func contains(sliceToSearch []string, searched string) bool {
 	return found
 }
 
+// jsonToCompactString transforms an unmarshalled JSON in form of a map of string->any to a plain string without any spacing.
+func jsonToCompactString(inputJson map[string]interface{}) (string, error) {
+	rawJson, err := json.Marshal(inputJson)
+	if err != nil {
+		return "", err
+	}
+	compactedJson := new(bytes.Buffer)
+	err = json.Compact(compactedJson, rawJson)
+	if err != nil {
+		return "", err
+	}
+	return compactedJson.String(), nil
+}
+
+// areMarshaledJsonEqual compares that two marshaled JSON strings are equal or not. Returns an error if something
+// wrong happens when compacting both for comparison.
```

Comment:
We are actually not compacting the two items, but encoding them

Reply:
True, that was some sort of leftover that I forgot to change. Thanks for spotting it!